### ISSUE TITLE
Custom partyId value validation

### ISSUE THREAD

**Issue by Benjamin23** [warappa/OpenTransSharp#23]
Hello, we have a custom PARTY_ID type that we use for our new customers. The value looks like this: 

`<bmecat:PARTY_ID type="new_customer">NewCustomerid</bmecat:PARTY_ID>`

We use this in the Order document, however, we are getting the following error when validating this value: 

`"Document validation failed!. \"/ORDER/ORDER_HEADER[1]/ORDER_INFO[1]/PARTIES[1]/PARTY[2]/bmecat:PARTY_ID[2][type]\"=[\"The 'type' attribute is invalid - The value 'new_customer' is invalid according to its datatype 'String' - The Pattern constraint failed.\"]`

When looking at the documentation, the validation rules are a list of 7 predefined values (buyer_specific, customer_specific, duns, iln, gln, party_specific and supplier_specific) and the user defined value with the format \w{1,250}. 

According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_class_escape#w) the value new_customer should pass this validation but we are still getting the error. 

Are we maybe missing something to support these custom values? If we test with the value `newcustomer ` then the validation passes, from the looks of it the `_` character is causing the problem, but it shouldn't matter since `\w` covers the `_` character also.

**Comment by warappa** [comment]
This seems to be caused by how **_XSD patterns_ (note: not "regexes") differ from regular regex rules**.

### TL;DR
`\w` in XSD pattern does not include punctation - which also means `_` - although in regex it would be included.

### Details
This question (and comments) exactly cover your issue with the underscore:
https://stackoverflow.com/questions/75277396/xml-schema-pattern-restriction-what-is-w

This explains it more clearly in an answer:
https://stackoverflow.com/questions/3367205/is-something-wrong-with-my-regex

**Comment by Benjamin23** [comment]
Thank you for the help, in that case, we'll find a workaround in our code.
